azure/azure-iot-hub-device-simulator/src/main/java/org/eclipse/examples/azure/DittoAzureIoTHubClient.java

```diff
@@ -0,0 +1,237 @@
+                        .build())
+                .build();
+    }
+
+    private static final class EventCallback implements IotHubEventCallback {
+
+        private static final Logger LOGGER = LoggerFactory.getLogger(EventCallback.class.getSimpleName());
+
+        @Override
+        public void execute(final IotHubStatusCode iotHubStatusCode, final Object o) {
+            LOGGER.info("Published Event and got status: <{}>.", iotHubStatusCode);
+        }
+    }
+
+    private static class MessageCallback implements com.microsoft.azure.sdk.iot.device.MessageCallback {
+
+        private static final Logger LOGGER = LoggerFactory.getLogger(MessageCallback.class.getSimpleName());
+
+        @Override
+        public IotHubMessageResult execute(final Message message, final Object o) {
+            LOGGER.info("Received message with content: <{}>.", serializeByteMessage(message.getBytes()));
+
+            return IotHubMessageResult.COMPLETE;
+        }
+
+        private static String serializeByteMessage(final byte[] bytes) {
+            return Base64.getEncoder().encodeToString(bytes);
+        }
+
+    }
+
+    private static class MethodCallback implements DeviceMethodCallback {
+
+        private static final Logger LOGGER = LoggerFactory.getLogger(MethodCallback.class.getSimpleName());
+
+        @Override
+        public DeviceMethodData call(final String methodName, final Object methodData, final Object context) {
+            LOGGER.info("Received invocation of method: <{}>, with body: <{}>", methodName, methodData);
+
+            if (methodName.equals("reboot")) {
+                tryRebootDevice((DeviceClient) context);
+                return new DeviceMethodData(HttpStatusCode.OK.toInt(), "Reboot triggered.");
+            }
+            return new DeviceMethodData(HttpStatusCode.BAD_REQUEST.toInt(), "No such method.");
+        }
+
+        private static void tryRebootDevice(final DeviceClient client) {
+            try {
+                rebootDevice(client);
+            } catch (final IOException e) {
+                LOGGER.error("Received error, while rebooting the device", e);
+            }
+        }
+
+        private static void rebootDevice(final DeviceClient client) throws IOException {
+            LOGGER.info("Initiating reboot.");
+            client.closeNow();
+            LOGGER.info("Closed client.");
+            client.open();
+            LOGGER.info("Reopened client.");
+        }
+    }
+
+    private static final class ConnectionStatusLogger implements IotHubConnectionStatusChangeCallback {
+
+        private static final Logger LOGGER = LoggerFactory.getLogger(ConnectionStatusLogger.class.getSimpleName());
+
+        @Override
+        public void execute(final IotHubConnectionStatus iotHubConnectionStatus,
+                final IotHubConnectionStatusChangeReason iotHubConnectionStatusChangeReason, final Throwable throwable,
```

Comment:
`throwable` should be in its own line (argument wrapping if more than three argument that don't fit into a single line).